File: ts/640/557.ts
function reverseWords(s: string): string {
  const cs: string[] = []
  let l = 0, r = -1
  for (const c of s) {
    if (c === ' ') {
      while (l < r) {
        const tmp = cs[r]
        cs[r--] = cs[l]
        cs[l++] = tmp
      }
      r = cs.length
      l = r + 1
    }
    cs.push(c)
    r += 1
  }

  while (l < r) {
    const tmp = cs[r]
    cs[r--] = cs[l]
    cs[l++] = tmp
  }
  return cs.join('')
};

File: c/640/557.c
#include <string.h>

char * reverseWords(char * s){
  int len = strlen(s);
  int i = 0;

  while (i < len) {
    int start = i;
    while (i < len && s[i] != ' ') i++;

    int l = start, r = i - 1;
    while (l < r) {
      char tmp = s[r];
      s[r--] = s[l];
      s[l++] = tmp;
    }

    while (i < len && s[i] == ' ') i++;
  }

  return s;
}